### PR TITLE
(PDB-3311) Improve command processing exception handling

### PR DESCRIPTION
**Pull request by rbrw**

### CODE REVIEW

File: src/puppetlabs/puppetdb/random.clj [hunk 1]
```diff
@@ -1,38 +1,35 @@
...
   "Generate a random string of optional length, only lower case alphabet chars"
   ([] (random-string (inc (rand-int 10))))
   ([length]
-     {:pre  [(integer? length)
-             (pos? length)]
-      :post [(string? %)
-             (= length (count %))]}
-     (let [ascii-codes (concat (range 97 123))]
-       (apply str (repeatedly length #(char (rand-nth ascii-codes)))))))
+   ;;(assert (and (int? length) (pos? length))) ;; Q: do we really need this?
```
Comment (austb) [inline src/puppetlabs/puppetdb/random.clj:26]: I'm fine with omitting the assertions.

File: test/puppetlabs/puppetdb/scf/storage_test.clj [hunk 2]
```diff
@@ -840,6 +842,24 @@
...
+  (let [bad-resource {:type "Class"
+                      :title (str/join "" (repeat 1000000 "yo"))
+                      :line 1337
+                      :exported false
+                      :file "badfile.txt"}]
+    (try
+      (replace-catalog!
+       (update catalog :resources
+               assoc {:type "Class" :title "updog"} bad-resource))
+      (is false)
```
Comment (austb) [inline test/puppetlabs/puppetdb/scf/storage_test.clj:856]: This is being flagged by eastwood
```
2021-04-15T22:04:03.2775864Z test/puppetlabs/puppetdb/scf/storage_test.clj:856:7: constant-test: Test expression is always logical true or always logical false: false in form (if value__9681__auto__ (clojure.test/do-report {:type :pass, :expected (quote false), :actual value__9681__auto__, :message nil}) (clojure.test/do-report {:type :fail, :expected (quote false), :actual value__9681__auto__, :message nil}))
```

File: src/puppetlabs/puppetdb/random.clj [hunk 3]
```diff
@@ -1,38 +1,35 @@
...
   (:require
    [clojure.string :as string]
-   [clojure.walk :refer [keywordize-keys]]))
+   [clojure.walk :refer [keywordize-keys]])
+  (:import
+   (org.apache.commons.lang3 RandomStringUtils)))
+
+;; FIXME: https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/RandomStringUtils.html
+
+(set! *warn-on-reflection* true)
```
Comment (austb) [inline src/puppetlabs/puppetdb/random.clj:10]: Should we remove this warn-on-reflections setting?

And should we remove the FIXME above as well, or provide some more context for what needs to be fixed?

**Review by austb — COMMENTED** [review]
After filtering out the reflection warnings I noticed a few new eastwood message that don't appear to be in the current output of `6.x`. We should see if they go away when we remove the reflection warnings and fix the test case.

```
2021-04-15T22:03:15.4368234Z Performance warning, puppetlabs/puppetdb/testutils/queue.clj:78:25 - case has int tests, but tested expression is not primitive.
...
2021-04-15T22:03:37.5235565Z combinatorics.clj:312 recur arg for primitive local: acc is not matching primitive, had: Object, needed: long
2021-04-15T22:03:37.5238962Z Auto-boxing loop arg: acc
2021-04-15T22:03:37.5244889Z combinatorics.clj:520 recur arg for primitive local: y is not matching primitive, had: Object, needed: long
2021-04-15T22:03:37.5247597Z Auto-boxing loop arg: y
2021-04-15T22:03:37.5250052Z combinatorics.clj:613 recur arg for primitive local: index is not matching primitive, had: Object, needed: long
2021-04-15T22:03:37.5252682Z Auto-boxing loop arg: index
2021-04-15T22:03:37.5255107Z combinatorics.clj:621 recur arg for primitive local: index is not matching primitive, had: Object, needed: long
2021-04-15T22:03:37.5258351Z Auto-boxing loop arg: index

File: test/puppetlabs/puppetdb/scf/storage_test.clj (hunk 2 above)
Comment (rbrw) [reply test/puppetlabs/puppetdb/scf/storage_test.clj]: Ahh, right - changed it to a throw.

File: src/puppetlabs/puppetdb/random.clj (hunk 3 above)
Comment (rbrw) [reply src/puppetlabs/puppetdb/random.clj]: Thanks.  Both vestigial - removed.

**Review by austb — APPROVED** [review]

**Pull request merged**